packages/replay/src/replay.ts

```diff
@@ -872,4 +878,18 @@ export class ReplayContainer implements ReplayContainerInterface {
       }, rateLimitDuration);
     }
   }
+
+  /** Loads (or refreshes) the current session. Returns false if session is not recorded. */
+  private _loadAndCheckSession(expiry?: number): boolean {
+    this._loadSession(expiry);
+
+    // We know this is set, because it is always set in _loadSession
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    if (!this.session!.sampled) {
+      this.stop();
+      return false;
+    }
+
+    return true;
+  }
```

Comment:
What are the cases where we want to load session but not check sampled?

Reply:
I think on `start`, because there we do some more stuff (incl. also checking this). but maybe actually we can also just use this there, looking at it again - I'll unify it!